### DISCUSSION TITLE
Communication between modules

### DISCUSSION THREAD

**Discussion by SomedicSense** [zephyrproject-rtos/zephyr#58220] [Q&A]
Hey, 

we are currently looking into different approaches for enabling communication between application framework and modules, and interaction between modules. Our current solution consist of a simple work queue that handles work placed on the module. We are currently looking into ZBus where it is possible to subscribe and publish to a channel. We also looked into implementing RPC. 

Has anyone worked on something similar? How have you handled it? Does Zephyr recommend any approach?

Any tips and discussions would be great!

Best Regards,
H

**Comment by rodrigopex** [comment]
Hi, @SomedicSense, I was not assigned to this issue, and it got lost somehow. Please clarify what you mean by _application framework_ and _modules_. This _module_ is a logically separated code portion, or is it a _Zephyr Module_? The _application framework_ is the application logic, or do you mean something different? If you like to discuss more about ZBus, I invite you to talk via Discord on the ZBus channel https://discord.gg/BtvEttqc. 

I would suggest you ask about that on a related Discord channel. It is not common to see this community answering quickly by issue.
